runelite-client/src/main/java/net/runelite/client/plugins/lmsperformancetracker/LmsPerformanceTrackerOverlay.java

```diff
@@ -0,0 +1,99 @@
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package net.runelite.client.plugins.lmsperformancetracker;
+
+import java.awt.Color;
+import java.awt.Dimension;
+import java.awt.Graphics2D;
+import net.runelite.client.ui.overlay.Overlay;
+import net.runelite.client.ui.overlay.OverlayMenuEntry;
+import net.runelite.client.ui.overlay.OverlayPosition;
+import net.runelite.client.ui.overlay.OverlayPriority;
+import net.runelite.client.ui.overlay.components.LineComponent;
+import net.runelite.client.ui.overlay.components.PanelComponent;
+import net.runelite.client.ui.overlay.components.TitleComponent;
+import javax.inject.Inject;
+
+import static net.runelite.api.MenuAction.RUNELITE_OVERLAY_CONFIG;
+import static net.runelite.client.ui.overlay.OverlayManager.OPTION_CONFIGURE;
+
+public class LmsPerformanceTrackerOverlay extends Overlay
+{
+	private final PanelComponent panelComponent = new PanelComponent();
+	private final LmsPerformanceTrackerPlugin plugin;
+	private final LmsPerformanceTrackerConfig config;
+
+	@Inject
+	private LmsPerformanceTrackerOverlay(LmsPerformanceTrackerPlugin plugin, LmsPerformanceTrackerConfig config)
+	{
+		super(plugin);
+		this.plugin = plugin;
+		this.config = config;
+		setPosition(OverlayPosition.BOTTOM_RIGHT);
+		setPriority(OverlayPriority.LOW);
+		panelComponent.setWrapping(4);
+		getMenuEntries().add(new OverlayMenuEntry(RUNELITE_OVERLAY_CONFIG, OPTION_CONFIGURE, "LMS Performance overlay"));
+	}
+
+	@Override
+	public Dimension render(Graphics2D graphics)
+	{
+		FightPerformance currentFight = plugin.getCurrentFight();
+		if (currentFight == null || !config.showCurrentFightOverlay() ||
+			(config.restrictToLms() && !plugin.isAtLMS()))
+		{
+			return null;
+		}
+
+		panelComponent.getChildren().clear();
+
+		// Only display the title if it's enabled (pointless in my opinion, since you can just see
+		// what the panel is displaying, but I can understand the preference of having overlays labelled)
+		if (config.showOverlayTitle())
+		{
+			panelComponent.getChildren().add(TitleComponent.builder()
+				.text("LMS Performance")
+				.build());
+		}
+
+		// First line: Player's stats (you: success rate counters + % value)
+		// Would have preferred to have usernames but they are too long.
```

Comment:
For the overlay, I'll make that change, good point. For the Panel, after using it a lot myself I personally prefer having both of them since they sort of give you different details. The fraction tells you how many total attacks there were while the percentage just gives you a quick look at your overall ratio. In some cases, an opponent could have a lot less attacks but with a high success rate. Depending on the attack speeds of weapons used, this can be an indicator of worse performance.